atlasdb-api/src/main/java/com/palantir/atlasdb/keyvalue/api/Cell.java

```diff
@@ -62,10 +67,21 @@ public static boolean isNameValid(byte[] name) {
         return name != null && name.length > 0 && name.length <= MAX_NAME_LENGTH;
     }
 
-    public static void validateNameValid(byte[] name) {
+    private void validateNameValid(byte[] name) {
         Preconditions.checkNotNull(name, "name cannot be null");
         Preconditions.checkArgument(name.length > 0, "name must be non-empty");
-        Preconditions.checkArgument(name.length <= MAX_NAME_LENGTH, "name must be no larger than " + MAX_NAME_LENGTH);
+
+        try {
+            Preconditions.checkArgument(
+                    name.length <= MAX_NAME_LENGTH,
+                    "name must be no longer than {}.",
+                    MAX_NAME_LENGTH);
+        } catch (IllegalArgumentException e) {
+            log.debug("Cell creation that was attempted was: {}; since the vast majority of people encountering this "
+                    + "problem are using unbounded Strings as components, it may aid your debugging to know the ASCII "
+                    + "interpretation of the bad field was: [{}]", this, new String(name, StandardCharsets.US_ASCII));
```

Comment:
Should probably use UTF-8 as that is what we encode to in the majority of places.